server/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -671,7 +671,7 @@ public boolean isRelocationSourceOf(ShardRouting other) {
     /** returns true if the current routing is identical to the other routing in all but meta fields, i.e., unassigned info */
     public boolean equalsIgnoringMetadata(ShardRouting other) {
         return primary == other.primary
-            && Objects.equals(shardId, other.shardId)
+            && shardId.equals(other.shardId)
```

Comment:
This is never `null` so no need for the extra indirection that (weirdly enough) doesn't inline here.